MaterialLib/SolidModels/SolidsBase.h

```diff
@@ -0,0 +1,87 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef MATERIALLIB_SOLIDMODELS_SOLIDBASE_H_
+#define MATERIALLIB_SOLIDMODELS_SOLIDBASE_H_
+
+#include <memory>
+
+#include "ProcessLib/Deformation/BMatrixPolicy.h"
+#include "ProcessLib/Parameter/Parameter.h"
+
+namespace MeshLib
+{
+class Element;
+}
+
+namespace Solids
+{
+template <int DisplacementDim>
+struct SolidsBase
+{
+    struct InternalVariables
+    {
+        virtual void pushBackState() = 0;
+    };
+
+    using KelvinVector = ProcessLib::KelvinVectorType<DisplacementDim>;
+    using KelvinMatrix = ProcessLib::KelvinMatrixType<DisplacementDim>;
+
+    // TODO Avoid copies of data:
+    // Using MatrixBase<Derived> not possible because template functions cannot
+    // be virtual. Maybe there is a workaround for this.
+    // Using Map<Matrix<double, ...>> makes the interface (for the material
+    // model implementation) unnecessary difficult.
+    void computeConstitutiveRelation(
+        double const t,
+        ProcessLib::SpatialPosition const& x,
+        double const dt,
+        Eigen::Matrix<double, -1, 1> const& eps_prev,
+        Eigen::Matrix<double, -1, 1> const& eps,
+        Eigen::Matrix<double, -1, 1> const& sigma_prev,
+        Eigen::Matrix<double, -1, 1>& sigma,
```

Comment:
maybe I'm missing a full picture... does it make sense to have two separate functions to get stress and C? in newton raphson, we evaluate residual and Jacobian in separate functions. hence we don't need both stress and C at the same time.